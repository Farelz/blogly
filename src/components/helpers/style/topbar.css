/******* TopBar Component ******/
.navbar-wrapper .topbar {
    background: hsl(36, 98%, 46%);
}

.navbar-wrapper .topbar .brand-logo {
    position: inherit;
    display: initial;
}

.navbar-wrapper .topbar .blog-info {
    display: inline-block;
}
.navbar-wrapper .topbar .blog-title {
    font-size: 21.516px;
    padding-left: 10px;
    padding-top: 9px;
    color: hsl(0, 0%, 14%);
}

.navbar-wrapper .topbar .social {
    margin-left: 5px;
    display: inline-block;
}

.navbar-wrapper .topbar .social a {
    display: inline-block;
    padding: 0 10px;
}

.navbar-wrapper .topbar .social i {
    font-size: 19px;
    color: rgba(255, 255, 255, 0.8);
}

.navbar-wrapper .topbar .nav-wrapper .user-image {
    width: 50px;
}

.navbar-wrapper .topbar .nav-wrapper a.user-profile {
    margin-top: 10%;
}

.navbar-wrapper .topbar .nav-wrapper li a {
    color: rgba(255, 255, 255, 0.8);
}

.navbar-wrapper .topbar .nav-wrapper li a.btn {
    background-color: transparent;
    border-radius: 3px;
    color: rgba(255, 255, 255, 0.8);
    border: 1.4px solid rgba(255, 255, 255, 0.8);
}

.navbar-wrapper .topbar .nav-content .menu {
    overflow-x: scroll;
    overflow-y: hidden;
    white-space: nowrap;
    -webkit-overflow-scrolling: touch;
}

.navbar-wrapper .topbar .nav-content .menu::-webkit-scrollbar {
    display: none;
}

.navbar-wrapper .topbar .nav-content .menu .item {
    display: inline-block;
    padding: 0 10px;
}

.navbar-wrapper .topbar .nav-content .nav-cta h5 {
    border-left: 1px solid rgba(255, 255, 255, 0.4);
    padding-left: 10px;
}

.navbar-wrapper .topbar .nav-content .menu .item h5 a,
.navbar-wrapper .topbar .nav-content .nav-cta h5 a {
    text-transform: uppercase;
    font-size: 15px;
    color: rgba(255, 255, 255, 0.8);
}

.navbar-wrapper .topbar ul a:hover {
    background-color: inherit;
}

/* Large Screen */

@media screen and (min-width: 993px) {
    .navbar-wrapper .topbar .nav-wrapper,
    .navbar-wrapper .topbar .nav-content {
        margin-left: 10%;
        margin-right: 7%;
    }
    .navbar-wrapper .topbar .nav-wrapper .user-image {
        width: 50px;
    }
    .navbar-wrapper .topbar .nav-wrapper a.user-profile {
        margin-top: 10%;
    }
    .navbar-wrapper .topbar .nav-content .menu {
        margin-top: -40px;
    }
    .navbar-wrapper .topbar .nav-content .nav-cta {
        margin-top: -40px;
    }
}

/* medium screen */
@media screen and (max-width: 992px) and (min-width: 601px) {
    .navbar-wrapper .topbar .nav-wrapper,
    .navbar-wrapper .topbar .nav-content {
        margin-left: 10%;
        margin-right: 7%;
    }
    .navbar-wrapper .topbar .nav-wrapper ul i {
        padding-top: 0px !important;
    }
}

/* Small Screen */

@media screen and (max-width: 600px) {
    .navbar-wrapper .topbar .nav-wrapper,
    .navbar-wrapper .topbar .nav-content {
        margin-left: 5%;
        margin-right: 1%;
    }
    .navbar-wrapper .topbar .nav-wrapper .brand-logo i {
        margin-top: 14px;
    }
    .navbar-wrapper .topbar .blog-title {
        border: none;
        padding-left: 10px;
        padding-top: 9px;
    }
    .navbar-wrapper .topbar .sidenav-trigger {
        margin-top: 6px;
        color:hsl(0, 0%, 14%);
    }
    .navbar-wrapper  ul.sidenav li a {
        color: #404040;
    }
    .navbar-wrapper ul.sidenav .user-view {
        background-color: hsl(0, 0%, 25%);
    }
    .navbar-wrapper ul.sidenav li a.email {
        padding-top: 20px;
    }
    .navbar-wrapper .topbar .nav-wrapper .brand-logo {
        font-size: 35px;
        left: 0%;
        transform: translateX(7%);
    }
    .navbar-wrapper .topbar .nav-wrapper .hide-small {
        display: none;
    }

    .navbar-wrapper .topbar .nav-wrapper ul i {
        font-size: 19px;
        padding-top: 15px;
    }

    .navbar-wrapper .topbar .nav-wrapper .search {
        padding: 7px 0;
    }

    .navbar-wrapper .topbar .nav-wrapper a.hide-on-med-and-up {
        padding: 6px 9px;
        color: #ffff;
    }

    .navbar-wrapper .topbar .nav-title {
        left: 50%;
        transform: translateX(9%);
        padding: 2px 0;
    }
    .navbar-wrapper .topbar .nav-content .m12,
    .navbar-wrapper .topbar .nav-content .nav-menu {
        padding: 0px;
    }
    .navbar-wrapper .topbar .social {
        display: none;
    }
    .navbar-wrapper .topbar .nav-content .nav-cta {
        display: inline;
    }
}
