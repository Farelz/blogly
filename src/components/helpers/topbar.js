/**
 * ./src/components/post/view/topbar
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class TopBar extends Component {
    render() {
        return (
            <nav className="topbar nav-extended">
                <div className="nav-wrapper">
                    <Link to="/" className="brand-logo">
                        <i className="fa fa-codiepie" />
                    </Link>
                    <div className="blog-info">
                        <span className="blog-title">freeCodeCamp</span>
                        <div className="social">
                            <a title="Twitter">
                                <i className="fa fa-twitter" />
                            </a>
                            <a title="Facebook">
                                <i className="fa fa-facebook" />
                            </a>
                        </div>
                    </div>
                    {this.props.user_data != undefined ? (
                        <ul className="right hide-on-small-only">
                            <li>
                                <a className="user-profile">
                                    <img
                                        className="user-image responsive-img circle"
                                        src={
                                            this.props.user_data.user
                                                .gravatarUrl
                                        }
                                    />
                                </a>
                            </li>
                        </ul>
                    ) : (
                        <ul className="right hide-on-small-only">
                            <li>
                                <Link to="/auth/login" className="sign-in">
                                    Sign in
                                </Link>
                            </li>
                            <li>
                                <Link to="/auth/signup" className="sign-up btn">
                                    Get started
                                </Link>
                            </li>
                        </ul>
                    )}
                    {this.props.user_data != undefined ? (
                        <ul className="right hide-on-med-and-up">
                            <li>
                                <a className="user-profile">
                                    <img
                                        className="user-image responsive-img circle"
                                        src={
                                            this.props.user_data.user
                                                .gravatarUrl
                                        }
                                    />
                                </a>
                            </li>
                        </ul>
                    ) : (
                        <ul className="right hide-on-med-and-up">
                            <li>
                                <Link
                                    className="hide-on-med-and-up"
                                    to="/auth/login"
                                >
                                    <i className="fa fa-sign-in" />
                                </Link>
                            </li>
                            <li>
                                <Link
                                    className="hide-on-med-and-up"
                                    to="/auth/login"
                                >
                                    <i className="fa fa-user-plus" />
                                </Link>
                            </li>
                        </ul>
                    )}
                </div>
            </nav>
        );
    }
}

TopBar.propTypes = {
    user_data: PropTypes.object
};

export default TopBar;
