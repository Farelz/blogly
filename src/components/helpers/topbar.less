/******* TopBar Component ******/
@primary_color: #1679ca;
@menu_color: #292929;

.nav-icon {
    font-size: 1.7em;
    margin-top: -1px;

    @media screen and (max-width: 992px) and (min-width: 601px) {
        margin-top: -3px;
    }
}

.navbar-wrapper {
    .topbar {
        background: hsl(0, 0%, 99.1%);
        box-shadow: 0px 0px 0px 0px hsl(0, 0%, 99.1%);

        .nav-wrapper {
            margin-left: 10%;
            margin-right: 7%;

            @media screen and (max-width: 600px) {
                margin-left: 5%;
                margin-right: 1%;

                .sidenav-trigger {
                    font-size: 25px;
                    margin: 1px 0px 1px 2px;
                    color: #4d4d4d !important;

                }
            }

            .brand-logo {
                position: inherit;
                display: inline-block;
                margin-top: 2px;

                @media screen and (max-width: 600px) {
                    margin-top: 0px;
                    font-size: 40px;
                    left: 50%;
                    transform: translateX(-70%);

                    i {
                        margin-top: 14px;
                    }
                }

                .blog-title {
                    font-size: 30.516px;
                    font-weight: 700;
                    color: @menu_color;
                    letter-spacing: -2.1px;

                    @media screen and (max-width: 600px) {
                        border: none;
                        padding-left: 10px;
                        padding-top: 9px;
                    }
                }
            }

            .notification-icon-on-small {
                .nav-icon;

                font-size: 1.6em;
                margin-top: 1px;
                padding-right: 10px;
                color: #4d4d4d;
            }

            ul {
                i {
                    @media screen and (max-width: 992px) and (min-width: 601px) {
                        padding-top: 0px !important;
                    }
                }

                li a {
                    color: @menu_color + hsl(0, 0%, 44%);
                    font-size: 15.11px;
                    margin-top: 10px;

                    &:hover {
                        color: @menu_color;
                        background-color: inherit;
                    }

                    &.btn {
                        background-color: transparent;
                        border-radius: 3px;
                        color: #fff;
                        border: 1.4px solid #fff;
                    }

                    &.notification-icon {
                        .nav-icon
                    }

                    &.user-profile {
                        margin-top: 10%;

                        .user-image {
                            width: 35px;
                            margin-top: 10px;

                            @media screen and (max-width: 992px) and (min-width: 601px) {
                                width: 30px;
                            }

                            &.defImg {
                                background-color: hsl(0, 0%, 90%);
                            }
                        }
                    }
                }

                .dropdown-container .dropdown-content {
                    border-radius: 3px;

                    li a {
                        color: #595959;
                        margin-top: 5px;
                        font-weight: 500;
                    }

                    @media screen and (min-width: 993px) {
                        top: 60px !important;
                        left: 1007px !important;
                        width: 170px !important;
                    }

                    @media screen and (max-width: 992px) and (min-width: 601px) {
                        top: 82px !important;
                        left: 507px !important;
                        width: 167px !important;
                    }
                }
            }
        }

        .nav-content {
            padding: 0 10% 3% 10%;

            div {
                overflow-x: scroll;
                
                &::-webkit-scrollbar {
                    display: none;
                }

                span a {
                    color: #868686 !important;
                    font-size: 15px;
                    text-transform: uppercase;
                    letter-spacing: -0.5px;
                    padding-right: 14px;

                    &:hover {
                        background-color: transparent;
                        color: @menu_color;
                    }

                    @media screen and (min-width: 993px) {
                        font-size: 16px;
                    }
                }
            }
        }
    }

    ul.sidenav {
        li {
            .user-view {
                background-color: #0e66af;

                img.defImg {
                    background-color: hsl(0, 0%, 90%);
                }

                &.guest {
                    padding-top: 25px;
                }

                .email {
                    padding-top: 20px;
                    color: #fff;
                    font-size: 15.11px;
                    font-weight: 500;
                }
            }
            a {
                color: #333333;
            }
        }
    }
}
