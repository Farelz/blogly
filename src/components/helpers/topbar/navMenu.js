import React from 'react';
// import types from 'prop-types';

const menu_items = [
    'home',
    'trust',
    'culture',
    'tech',
    'entreprenuership',
    'self',
    'politics',
    'media',
    'design',
    'programming',
    'art',
    'science',
    'popular'
];

export const NavMenu = () => (
    <div>
        <section>
            {menu_items.map((item, id) => (
                <span key={id}>
                    <a>{item}</a>
                </span>
            ))}
        </section>
    </div>
);
