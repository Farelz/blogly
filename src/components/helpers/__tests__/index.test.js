import React from 'react';
import { shallow } from 'enzyme';

import { TopBar } from '../topbar';
import { NavBar } from '../topbar/navBar';

describe('topbar tests', () => {
    const logout = jest.fn();
    const readNotifications = jest.fn();
    const topbarProps = {
        isPostPage: false,
        user_data:{
            user: {}
        },
        notifications: {},
        logout,
        readNotifications,
    };

    it('renders correctly when logged out', () => {
        const wrapper = shallow(<TopBar {...topbarProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly when logged in', () => {
        const wrapper = shallow(<TopBar {...topbarProps} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe('nav bar unit tests', () => {
    const logout = jest.fn();
    const readNotifications = jest.fn();
    const Props = {
        isLoggedIn: false,
        isPostPage: false,
        stick: false,
        imageData: {
            image: '',
            imageClass: ''
        },
        user_data:{
            user: {}
        },
        notifications_data: {
            notifications: [],
            unread_count: 0
        },
        logout,
        readNotifications,
    };

    it('renders', () => {
        const wrapper = shallow(<NavBar {...Props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('shows user navbar when loggedin', () => {
        const props = {
            ...Props,
            isLoggedIn: true
        };
        const wrapper = shallow(<NavBar {...props}/>);
        expect(wrapper.find('[data-target="dropdown-menu"]')).toHaveLength(1);
        expect(wrapper.find('[data-target="notifications-menu"]')).toHaveLength(1);
        expect(wrapper.find('.sign-up')).toHaveLength(0);
        expect(wrapper.find('.sign-in')).toHaveLength(0);
    });

    it('shows user navbar when loggedin', () => {
        const wrapper = shallow(<NavBar {...Props}/>);
        expect(wrapper.find('[data-target="dropdown-menu"]')).toHaveLength(0);
        expect(wrapper.find('[data-target="notifications-menu"]')).toHaveLength(0);
        expect(wrapper.find('.sign-up')).toHaveLength(1);
        expect(wrapper.find('.sign-in')).toHaveLength(1);
    });
});
