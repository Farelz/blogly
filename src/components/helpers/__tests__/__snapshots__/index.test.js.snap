// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav bar unit tests renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    imageData={
      Object {
        "image": "",
        "imageClass": "",
      }
    }
    isLoggedIn={false}
    isPostPage={false}
    logout={[MockFunction]}
    notifications_data={
      Object {
        "notifications": Array [],
        "unread_count": 0,
      }
    }
    readNotifications={[MockFunction]}
    stick={false}
    user_data={
      Object {
        "user": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="nav-wrapper"
        >
          <a
            className="sidenav-trigger hide-on-med-and-up"
            data-target="mobile-topbar"
            href="#"
          >
            <MdMenu />
          </a>
          <Link
            className="brand-logo center"
            replace={false}
            to="/"
          >
            <span
              className="blog-title"
            >
              ReactPress
            </span>
          </Link>
          <ul
            className="right hide-on-small-only"
          >
            <React.Fragment>
              <li>
                <Link
                  className="sign-in"
                  replace={false}
                  to="/auth/login"
                >
                  Sign in
                </Link>
              </li>
              <li>
                <Link
                  className="sign-up btn"
                  replace={false}
                  to="/auth/signup"
                >
                  Get started
                </Link>
              </li>
            </React.Fragment>
          </ul>
        </div>,
        false,
      ],
      "className": "topbar nav-extended",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="sidenav-trigger hide-on-med-and-up"
              data-target="mobile-topbar"
              href="#"
            >
              <MdMenu />
            </a>,
            <Link
              className="brand-logo center"
              replace={false}
              to="/"
            >
              <span
                className="blog-title"
              >
                ReactPress
              </span>
            </Link>,
            <ul
              className="right hide-on-small-only"
            >
              <React.Fragment>
                <li>
                  <Link
                    className="sign-in"
                    replace={false}
                    to="/auth/login"
                  >
                    Sign in
                  </Link>
                </li>
                <li>
                  <Link
                    className="sign-up btn"
                    replace={false}
                    to="/auth/signup"
                  >
                    Get started
                  </Link>
                </li>
              </React.Fragment>
            </ul>,
            false,
          ],
          "className": "nav-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MdMenu />,
              "className": "sidenav-trigger hide-on-med-and-up",
              "data-target": "mobile-topbar",
              "href": "#",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span
                className="blog-title"
              >
                ReactPress
              </span>,
              "className": "brand-logo center",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ReactPress",
                "className": "blog-title",
              },
              "ref": null,
              "rendered": "ReactPress",
              "type": "span",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <React.Fragment>
                <li>
                  <Link
                    className="sign-in"
                    replace={false}
                    to="/auth/login"
                  >
                    Sign in
                  </Link>
                </li>
                <li>
                  <Link
                    className="sign-up btn"
                    replace={false}
                    to="/auth/signup"
                  >
                    Get started
                  </Link>
                </li>
              </React.Fragment>,
              "className": "right hide-on-small-only",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <li>
                    <Link
                      className="sign-in"
                      replace={false}
                      to="/auth/login"
                    >
                      Sign in
                    </Link>
                  </li>,
                  <li>
                    <Link
                      className="sign-up btn"
                      replace={false}
                      to="/auth/signup"
                    >
                      Get started
                    </Link>
                  </li>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="sign-in"
                      replace={false}
                      to="/auth/login"
                    >
                      Sign in
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign in",
                      "className": "sign-in",
                      "replace": false,
                      "to": "/auth/login",
                    },
                    "ref": null,
                    "rendered": "Sign in",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="sign-up btn"
                      replace={false}
                      to="/auth/signup"
                    >
                      Get started
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Get started",
                      "className": "sign-up btn",
                      "replace": false,
                      "to": "/auth/signup",
                    },
                    "ref": null,
                    "rendered": "Get started",
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": Symbol(react.fragment),
            },
            "type": "ul",
          },
          false,
        ],
        "type": "div",
      },
      false,
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="nav-wrapper"
          >
            <a
              className="sidenav-trigger hide-on-med-and-up"
              data-target="mobile-topbar"
              href="#"
            >
              <MdMenu />
            </a>
            <Link
              className="brand-logo center"
              replace={false}
              to="/"
            >
              <span
                className="blog-title"
              >
                ReactPress
              </span>
            </Link>
            <ul
              className="right hide-on-small-only"
            >
              <React.Fragment>
                <li>
                  <Link
                    className="sign-in"
                    replace={false}
                    to="/auth/login"
                  >
                    Sign in
                  </Link>
                </li>
                <li>
                  <Link
                    className="sign-up btn"
                    replace={false}
                    to="/auth/signup"
                  >
                    Get started
                  </Link>
                </li>
              </React.Fragment>
            </ul>
          </div>,
          false,
        ],
        "className": "topbar nav-extended",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="sidenav-trigger hide-on-med-and-up"
                data-target="mobile-topbar"
                href="#"
              >
                <MdMenu />
              </a>,
              <Link
                className="brand-logo center"
                replace={false}
                to="/"
              >
                <span
                  className="blog-title"
                >
                  ReactPress
                </span>
              </Link>,
              <ul
                className="right hide-on-small-only"
              >
                <React.Fragment>
                  <li>
                    <Link
                      className="sign-in"
                      replace={false}
                      to="/auth/login"
                    >
                      Sign in
                    </Link>
                  </li>
                  <li>
                    <Link
                      className="sign-up btn"
                      replace={false}
                      to="/auth/signup"
                    >
                      Get started
                    </Link>
                  </li>
                </React.Fragment>
              </ul>,
              false,
            ],
            "className": "nav-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MdMenu />,
                "className": "sidenav-trigger hide-on-med-and-up",
                "data-target": "mobile-topbar",
                "href": "#",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span
                  className="blog-title"
                >
                  ReactPress
                </span>,
                "className": "brand-logo center",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ReactPress",
                  "className": "blog-title",
                },
                "ref": null,
                "rendered": "ReactPress",
                "type": "span",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <React.Fragment>
                  <li>
                    <Link
                      className="sign-in"
                      replace={false}
                      to="/auth/login"
                    >
                      Sign in
                    </Link>
                  </li>
                  <li>
                    <Link
                      className="sign-up btn"
                      replace={false}
                      to="/auth/signup"
                    >
                      Get started
                    </Link>
                  </li>
                </React.Fragment>,
                "className": "right hide-on-small-only",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <li>
                      <Link
                        className="sign-in"
                        replace={false}
                        to="/auth/login"
                      >
                        Sign in
                      </Link>
                    </li>,
                    <li>
                      <Link
                        className="sign-up btn"
                        replace={false}
                        to="/auth/signup"
                      >
                        Get started
                      </Link>
                    </li>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        className="sign-in"
                        replace={false}
                        to="/auth/login"
                      >
                        Sign in
                      </Link>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Sign in",
                        "className": "sign-in",
                        "replace": false,
                        "to": "/auth/login",
                      },
                      "ref": null,
                      "rendered": "Sign in",
                      "type": [Function],
                    },
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        className="sign-up btn"
                        replace={false}
                        to="/auth/signup"
                      >
                        Get started
                      </Link>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Get started",
                        "className": "sign-up btn",
                        "replace": false,
                        "to": "/auth/signup",
                      },
                      "ref": null,
                      "rendered": "Get started",
                      "type": [Function],
                    },
                    "type": "li",
                  },
                ],
                "type": Symbol(react.fragment),
              },
              "type": "ul",
            },
            false,
          ],
          "type": "div",
        },
        false,
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`topbar tests renders correctly when logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopBar
    isPostPage={false}
    logout={[MockFunction]}
    notifications={Object {}}
    readNotifications={[MockFunction]}
    user_data={
      Object {
        "user": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavBar
          imageData={
            Object {
              "image": undefined,
              "imageClass": "",
            }
          }
          isLoggedIn={true}
          isPostPage={true}
          logout={[Function]}
          notifications_data={Object {}}
          readNotifications={[Function]}
          stick={false}
        />,
        <SideNav
          data={
            Object {
              "image": undefined,
              "imageClass": "",
              "isLoggedIn": true,
              "username": undefined,
            }
          }
          logout={[Function]}
        />,
      ],
      "className": "navbar-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "imageData": Object {
            "image": undefined,
            "imageClass": "",
          },
          "isLoggedIn": true,
          "isPostPage": true,
          "logout": [Function],
          "notifications_data": Object {},
          "readNotifications": [Function],
          "stick": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "image": undefined,
            "imageClass": "",
            "isLoggedIn": true,
            "username": undefined,
          },
          "logout": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavBar
            imageData={
              Object {
                "image": undefined,
                "imageClass": "",
              }
            }
            isLoggedIn={true}
            isPostPage={true}
            logout={[Function]}
            notifications_data={Object {}}
            readNotifications={[Function]}
            stick={false}
          />,
          <SideNav
            data={
              Object {
                "image": undefined,
                "imageClass": "",
                "isLoggedIn": true,
                "username": undefined,
              }
            }
            logout={[Function]}
          />,
        ],
        "className": "navbar-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "imageData": Object {
              "image": undefined,
              "imageClass": "",
            },
            "isLoggedIn": true,
            "isPostPage": true,
            "logout": [Function],
            "notifications_data": Object {},
            "readNotifications": [Function],
            "stick": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "image": undefined,
              "imageClass": "",
              "isLoggedIn": true,
              "username": undefined,
            },
            "logout": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`topbar tests renders correctly when logged out 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopBar
    isPostPage={false}
    logout={[MockFunction]}
    notifications={Object {}}
    readNotifications={[MockFunction]}
    user_data={
      Object {
        "user": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavBar
          imageData={
            Object {
              "image": undefined,
              "imageClass": "",
            }
          }
          isLoggedIn={true}
          isPostPage={true}
          logout={[Function]}
          notifications_data={Object {}}
          readNotifications={[Function]}
          stick={false}
        />,
        <SideNav
          data={
            Object {
              "image": undefined,
              "imageClass": "",
              "isLoggedIn": true,
              "username": undefined,
            }
          }
          logout={[Function]}
        />,
      ],
      "className": "navbar-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "imageData": Object {
            "image": undefined,
            "imageClass": "",
          },
          "isLoggedIn": true,
          "isPostPage": true,
          "logout": [Function],
          "notifications_data": Object {},
          "readNotifications": [Function],
          "stick": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "image": undefined,
            "imageClass": "",
            "isLoggedIn": true,
            "username": undefined,
          },
          "logout": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavBar
            imageData={
              Object {
                "image": undefined,
                "imageClass": "",
              }
            }
            isLoggedIn={true}
            isPostPage={true}
            logout={[Function]}
            notifications_data={Object {}}
            readNotifications={[Function]}
            stick={false}
          />,
          <SideNav
            data={
              Object {
                "image": undefined,
                "imageClass": "",
                "isLoggedIn": true,
                "username": undefined,
              }
            }
            logout={[Function]}
          />,
        ],
        "className": "navbar-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "imageData": Object {
              "image": undefined,
              "imageClass": "",
            },
            "isLoggedIn": true,
            "isPostPage": true,
            "logout": [Function],
            "notifications_data": Object {},
            "readNotifications": [Function],
            "stick": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "image": undefined,
              "imageClass": "",
              "isLoggedIn": true,
              "username": undefined,
            },
            "logout": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
